library/src/include/tensile_host.hpp

```diff
@@ -40,29 +40,40 @@ template <typename Ti, typename To = Ti, typename Tc = To>
 struct RocblasContractionProblem
 {
     rocblas_handle    handle;
-    rocblas_operation trans_a;
-    rocblas_operation trans_b;
+    rocblas_operation trans_a = rocblas_operation_none;
+    rocblas_operation trans_b = rocblas_operation_none;
 
     // The size data members should exactly match Tensile's size parameters
     // even if rocBLAS uses smaller or differently-signed types
-    size_t    m;
-    size_t    n;
-    size_t    k;
+    size_t m;
+    size_t n;
+    size_t k;
+
     const Tc* alpha;
+
     const Ti* A;
-    size_t    ld_a;
-    size_t    stride_a;
+    size_t    row_stride_a = 1;
+    size_t    col_stride_a;
+    size_t    batch_stride_a;
+
     const Ti* B;
-    size_t    ld_b;
-    size_t    stride_b;
+    size_t    row_stride_b = 1;
+    size_t    col_stride_b;
+    size_t    batch_stride_b;
+
     const Tc* beta;
+
     const To* C;
-    size_t    ld_c;
-    size_t    stride_c;
-    To*       D        = const_cast<To*>(C); // Default D = C
-    size_t    ld_d     = ld_c;
-    size_t    stride_d = stride_c;
-    size_t    batch_count;
+    size_t    row_stride_c = 1;
+    size_t    col_stride_c;
+    size_t    batch_stride_c;
+
+    To*    D              = const_cast<To*>(C); // Default D = C
+    size_t row_stride_d   = row_stride_c;
+    size_t col_stride_d   = col_stride_c;
+    size_t batch_stride_d = batch_stride_c;
```

Comment:
It simplifies the coding, by allowing `C==D` to be represented by not initializing `D`.

Reply:
If `D` is not initialized by the constructor, then this sets `D=C`. The members corresponding to `D` are always initialized after the members corresponding to `C`, because they appear later in the member list.

We could remove this and make every constructor initialize every member, but it would be more verbose.